Build issues with OneAPI

@amametjanov noticed that SCORPIO failed to build (while building the following E3SM test) on Chrysalis with the Intel OneAPI compiler,

```
./cime/scripts/create_test --compiler oneapi-ifx SMS_P12x1.ne4_oQU240.WCYCL1850NS
```
Excerpt from pio.bldlog* :
```
...
/gpfs/fs1/soft/chrysalis/spack-latest/opt/spack/linux-rhel8-x86_64/oneapi-2022.2.0/openmpi-4.1.3-yb6p67p/bin/mpicc -DCPRINTELLLVM -DLINUX -DLOGGING -DMPICH_SKIP_MPICXX -DNETCDF_C_LOGGING_ENABLED -DNETCDF_C_NC__ENDDEF_EXISTS -DOMPI_SKIP_MPICXX -DTIMING -DUSE_MPI_ISEND_FOR_FC -DUSE_PNETCDF_VARN -DUSE_PNETCDF_VARN_ON_READ -O2 -traceback -fp-model precise -DTIMING -DLINUX  -DFORTRANUNDERSCORE -DNO_R16 -DCPRINTEL -DHAVE_SLASHPROC -DHIDE_MPI -DNDEBUG -DHAVE_MPI -DMCT_INTERFACE -DPIO2 -DHAVE_SLASHPROC -D_PNETCDF -DATM_PRESENT -DICE_PRESENT -DLND_PRESENT -DOCN_PRESENT -DROF_PRESENT -DGLC_PRESENT -DWAV_PRESENT -DESP_PRESENT -DMED_PRESENT -DPIO2  ... -MD -MT src/clib/CMakeFiles/pioc.dir/pio_lists.c.o -MF CMakeFiles/pioc.dir/pio_lists.c.o.d -o CMakeFiles/pioc.dir/pio_lists.c.o -c /lcrc/group/e3sm/azamat/repos/E3SM-swing/externals/scorpio/src/clib/pio_lists.c
...
/lcrc/group/e3sm/azamat/repos/E3SM-swing/externals/scorpio/src/clib/pio_lists.c:199:19: error: call to undeclared function 'spio_write_file_io_summary'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
            ret = spio_write_file_io_summary(pf);
                  ^
1 error generated.
```

--- a/src/clib/pio_lists.c
+++ b/src/clib/pio_lists.c
@@ -11,6 +11,7 @@
 #include <string.h>
 #include <stdio.h>
 #include "spio_file_mvcache.h"
+#include "spio_io_summary.h"
 
 static io_desc_t *pio_iodesc_list = NULL;
 static io_desc_t *current_iodesc = NULL;

